src/pages/PassportRedemption/GiveawayPopover.tsx

```diff
@@ -0,0 +1,237 @@
+import React , { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import GreenCheck from './GreenCheck'
+import InstagramEnabled from './InstagramEnabled'
+import InstagramDisabled from './InstagramDisabled';
+
+import { getAllParticipatingSellers, getGiveawayTicketsForContact } from '../../utilities/api/interactionManager'
+
+
+interface Props {
+  showInstagram : boolean
+  contactId: string
+}
+
+interface TextProps {
+  icon ?: boolean
+}
+
+interface IgButtonProps {
+  disabled ?: boolean
+}
+interface GiveawayData {
+  weekly_giveaway_entries: String
+  number_of_visits_left : undefined | number
+}
+
+const GiveawayPopover = (props: Props) => {
+  const [showInfo, setShowInfo] = useState(false);
+  const [giveawayData, setData] = useState<GiveawayData>({
+    weekly_giveaway_entries: '',
+    number_of_visits_left: undefined
+  })
+  useEffect(() => {
+    async function setGiveawayCount(contactId: string) {
+      try {
+        const qtySellers = await getAllParticipatingSellers()
```

Comment:
nit: I think we should avoid mixing `await` and `.then()` logic together, and honestly I don't think it's needed here. You should be able to do:

```suggestion
        const qtySellers = await getAllParticipatingSellers().data.length
```

If it's a typing issue you can cast to `any`, use `!.`, or type the underlying function.

Same for the below call